public/app/types/unified-alerting-dto.ts

```diff
@@ -290,6 +292,7 @@ export type GrafanaRecordingRuleDefinition = GrafanaRuleDefinition & {
 export interface RulerGrafanaRuleDTO<T = GrafanaRuleDefinition> {
   grafana_alert: T;
   for?: string;
+  keep_firing_for?: string;
```

Comment:
The API seems to treat this field as optional, but in other places, especially in the form handling we seem to treat this field as required. Should we make that consistent and mark the field as optional everywhere?

Reply:
YES